src/web/Dockerfile

```diff
@@ -0,0 +1,57 @@
+# NEXT_PUBLIC_API_URL is needed by the next.js build, which we define
+# as a build ARG in API_URL
+ARG API_URL
+# API Service URLs, set via ENV in docker or next build
+ARG IMAGE_URL
+ARG AUTH_URL
+ARG POSTS_URL
+ARG SEARCH_URL
+ARG FEED_DISCOVERY_URL
+# Front-end Web URL, set via ENV in docker or next build
+ARG WEB_URL
+
+FROM node:lts-alpine as base
+
+# Tini Entrypoint for Alpine
+# util-linux required to optimize builds using multiple cores
+RUN apk add --no-cache tini util-linux
+ENTRYPOINT [ "/sbin/tini", "--"]
+# what is happening?
+WORKDIR /frontend
+
+COPY . .
+
+FROM base as dependencies
+
+COPY .git .git
```

Comment:
I wanted to ask about this, but also see that we're failing CI due to this:

```
Step 14/32 : COPY .git .git
COPY failed: file not found in build context or excluded by .dockerignore: stat .git: file does not exist
```

@raygervais why did we include `.git`, was it for getting the version SHA in the frontend?  I wonder if we could be more clever about how we do this so we don't jam the entire `.git` into the build context.  NOTE: other than fixing this failure, we don't need to solve this in this PR, but reading it made me think.

Reply:
That's correct, if memory serves me that was exactly the reason. Realistically, .git could be stored as a dedicated micro-service elsewhere which has it's own basic API for getting SHA, commit data etc. Would remove all the weight from frontend entirely (or, we write a script which produces a JSON that the frontend consumes.. but where's the fun in that?)

Reply:
We should probably pass this as an arg to the build script, since we know it in the deployment server, and remove from docker.

Reply:
> We should probably pass this as an arg to the build script, since we know it in the deployment server, and remove from docker.

I'm not really sure about how to do this. Would it make sense to extract the commit's sha from the webhook request's payload and pass it to docker via env var so it can be used while building telescope? something like:

```sh
GIT_COMMIT_SHA=98EEE45A830AA65F768EC194A091349A41113BD0 docker-compose -f config/production.yml...
```
Any ideas or suggestions are welcome.

Reply:
We could just set it in the env early via command?

```sh
GIT_COMMIT=$(git rev-parse HEAD)
docker-compose -e GIT_COMMIT -f config/production.yml...
```

Reply:
I think the front-end is fetching git info using the `/health` route (exactly [here](https://github.com/Seneca-CDOT/telescope/blob/65f148ad35727e3b43c74da70370e985e5727a1e/src/web/src/components/Banner.tsx#L149)), so I think not adding the `.git` folder to the build process should be enough.